src/features/views/components/MoveViewDialog.tsx

```diff
@@ -0,0 +1,266 @@
+  }
+
+  const doMove = (folderId: number | null) => {
+    moveItem(itemToMove.type, itemToMove.data.id, folderId);
+    close();
+  };
+
+  return (
+    <Dialog
+      fullScreen={fullScreen}
+      fullWidth
+      maxWidth={'sm'}
+      onClose={close}
+      open={true}
+      PaperProps={{
+        sx: {
+          height: '80vh',
+          maxHeight: '600px',
+        },
+      }}
+      scroll="paper"
+    >
+      <DialogTitle>
+        <MoveItemBreadcrumbs
+          onClickFolder={setViewedFolder}
+          orgId={orgId}
+          viewedFolder={viewedFolder}
+        />
+      </DialogTitle>
+
+      <DialogContent>
+        <ZUIFuture future={itemsFuture}>
+          {(data) => {
+            const relevantItems = data.filter(
+              // This explicit typing should not be necessary, but it is. Don't know why.
+              (item): item is ViewBrowserFolderItem | ViewBrowserViewItem =>
+                item.type != 'back'
+            );
+            if (relevantItems.length == 0) {
+              return (
+                <List>
+                  <ListItem
+                    sx={{
+                      color: theme.palette.onSurface.disabled,
+                    }}
+                  >
+                    {messages.moveViewDialog.emptyFolder()}
+                  </ListItem>
+                </List>
+              );
+            }
+            return (
+              <List>
+                {relevantItems.map((item) => {
+                  const {
+                    data: { id },
+                    title,
+                    type,
+                  } = item;
+
+                  return (
+                    <ListItem
+                      key={`${type}-${id}`}
+                      onClick={
+                        type == 'folder' ? () => setViewedFolder(id) : undefined
+                      }
+                      sx={
+                        type == 'folder'
+                          ? { cursor: 'pointer' }
+                          : {
+                              color: theme.palette.onSurface.disabled,
+                            }
+                      }
+                    >
+                      <Box alignItems="center" display="flex" width="100%">
+                        <Box
+                          alignItems="flex-start"
+                          display="flex"
+                          marginRight={2}
+                        >
+                          <BrowserItemIcon item={item} />
+                        </Box>
+
+                        <Box
+                          alignItems="center"
+                          display="flex"
+                          flexGrow={1}
+                          marginRight={2}
+                        >
+                          {type == 'folder' ? (
+                            <Link color="inherit" underline="hover">
+                              {title}
+                            </Link>
```

Comment:
Maybe this is where the link can be disabled if the folder is the item that is currently being moved.